docassemble/assemblylinewizard/test_map_names.py

```diff
@@ -85,52 +49,155 @@ def test_mapped_scenarios(self, run_from_yaml=False):
+  "guardians_ad_litem": "guardians_ad_litem",
+  "witnesses": "witnesses",
 
   # Starts with no names
   "docket_number": "docket_numbers[0]",
-  "docket_numbers": "str(docket_numbers)",
+  "docket_numbers": "docket_numbers",
+  "signature_date": "signature_date",
+
+  # Reserved start with unreserved end
+  "user_address_street2_zip": "user_address_street2_zip",  
+
+  # Not reserved
+  "my_user_name_last": "my_user_name_last",
+  "foo": "foo",
+}
+
+interview_order_scenarios = {
+  # Reserved whole words
+  "signature_date": "signature_date",
+  "attorney_of_record_address_on_one_line": "attorney_of_record_address_on_one_line",
+
+  # Reserved endings
+  "user1": "users.gather()",
+  "user2": "users.gather()",
+  "user__2": "users.gather()",
+  "user_name": "users.gather()",
+  "user_name_full": "users.gather()",
+  "user_name_first": "users.gather()",
+  "user_name_middle": "users.gather()",
+  "user_name_last": "users.gather()",
+  "user_name_suffix": "users.gather()",
+  "user_gender": "users[0].gender",
+  "user_birthdate": "users[0].birthdate",
+  "user_age": "users[0].birthdate",
+  "user_address_unit": "users[0].address",
+  "user_address_address": "users[0].address",
+  "user_address_city": "users[0].address",
+  "user_email": "users[0].email",
+  "user2_phone": "users[1].phone_number",
+  "user_signature": "users[0].signature",
+
+  # County
+  # "county_name_short": not implemented,
+  # "county_division": not implemented,
+  "court_address_county": "courts[0].address",
+  "court_county": "courts[0].address",
+
+  # # Reserved starts (with names)
+  "user": "users.gather()",
+  "users": "users.gather()",
+  "plaintiff": "plaintiffs.gather()",
+  "defendant": "defendants.gather()",
+  "courts": "courts.gather()",
+  "other_parties": "other_parties.gather()",
+  "children": "children.gather()",
+  "guardians_ad_litem": "guardians_ad_litem.gather()",
+  "witnesses": "witnesses.gather()",
+
+  "defendant1_name": "defendants.gather()",
+  "defendant1_email": "defendants[0].email",
+
+  # Starts with no names
+  "docket_number": "docket_numbers.gather()",
+  "docket_numbers": "docket_numbers.gather()",
   "signature_date": "signature_date",
 
   # Reserved start with unreserved end
-  "user_address_street2_zip": "users[0].address_street2_zip",
+  "user_address_street2_zip": "user_address_street2_zip",
 
   # Not reserved
   "my_user_name_last": "my_user_name_last",
   "foo": "foo",
 }
+
+
+class TestMapNames(unittest.TestCase):
+    def setUp(self):
+        pass
+
+    def test_mapped_scenarios(self, run_from_yaml=False):
+        test_scenarios = [
+          (attachment_scenarios, map_names),
+          (interview_order_scenarios, lambda x: trigger_gather_string(map_names(x)))
+        ]
+        # Look in the console for a prettier version of the messages
+        passed = []
+        errored = []
+        for scenarios, function in test_scenarios:
+          temp_passed, temp_errored = self.run_scenarios(scenarios, function)
```

Comment:
Future issue: This might be the place to store which test these results belong to